manifests/config.pp

```diff
@@ -22,15 +21,72 @@
     mode    => '0600',
   }
 
-  file {'/etc/httpd/conf.d/pulp.conf':
+  apache::vhost { 'pulp-http':
+    priority            => '05',
+    docroot             => '/srv/pulp',
+    port                => 80,
+    servername          => [$::fqdn],
+    serveraliases       => [$::hostname],
+    additional_includes => '/etc/pulp/vhosts80/*.conf',
+    add_default_charset => 'UTF-8',
+  }
+
+  apache::vhost {'pulp-https':
+    priority                   => '05',
+    docroot                    => '/srv/pulp',
+    port                       => 443,
+    servername                 => [$::fqdn],
+    serveraliases              => [$::hostname],
+    add_default_charset        => 'UTF-8',
+    ssl                        => true,
+    ssl_cert                   => $pulp::consumers_ca_cert,
+    ssl_key                    => $pulp::consumers_ca_key,
+    ssl_ca                     => $pulp::consumers_ca_cert,
+    ssl_verify_client          => 'optional',
+    ssl_protocol               => ' all -SSLv2',
+    ssl_options                => '+StdEnvVars +ExportCertData',
+    ssl_verify_depth           => '3',
+    wsgi_process_group         => 'pulp',
+    wsgi_application_group     => 'pulp',
+    wsgi_daemon_process        => 'pulp user=apache group=apache processes=1 threads=8 display-name=%{GROUP}',
+    wsgi_pass_authorization    => 'On',
+    wsgi_import_script         => '/srv/pulp/webservices.wsgi',
+    wsgi_import_script_options => {
+      'process-group'     => 'pulp',
+      'application-group' => 'pulp'
+    },
+    wsgi_script_aliases        => {
+      '/pulp/api' => '/srv/pulp/webservices.wsgi'
+    },
+    directories                => [
+      {
+        'path'     => 'webservices.wsgi',
+        'provider' => 'files',
+      },
+      {
+        'path'     => '/srv/pulp',
+        'provider' => 'directory',
+      }
+      ],
+    options                    => ['SymLinksIfOwnerMatch'],
+    custom_fragment            => '  SSLInsecureRenegotiation on'
```

Comment:
The comment (allow older yum clients to connect, see bz 647828) is helpful to understand why it's needed and I'd at least leave it here.